microprofile/health-smallrye/src/main/java/org/wildfly/extension/microprofile/health/HealthReporterService.java

```diff
@@ -85,7 +85,7 @@ public void start(StartContext context) {
         final boolean defaultServerProceduresDisabled = ConfigProvider.getConfig().getOptionalValue("mp.health.disable-default-procedures", Boolean.class).orElse(false);
         healthReporter = new HealthReporter(emptyLivenessChecksStatus, emptyReadinessChecksStatus, defaultServerProceduresDisabled);
 
-        modelControllerClient = modelControllerClientFactory.get().createClient(managementExecutor.get());
+        modelControllerClient = modelControllerClientFactory.get().createSuperUserClient(managementExecutor.get(), true);
```

Comment:
I haven't got time today to check into the code but I suspect this will be triggering a permissions check here.  

The normal reaction to that would be "I must add a doProvileged" but I think this code is correct as the complete call stack will be initiated from the management endpoint.

But assuming my two assumptions are true it may be worth adding a comment so no one helpfully adds a "missing doPrivileged" at a later point.

Reply:
@darranl I've added a comment based on your remark.

I've manually tested that if RBAC is enabled, this code works (when security-enabled is false).
When security-enabled is true, the authentication fails when the HTTP endpoint is accessed and this code is not invoked.